core/embed/rust/src/ui/model_tt/component/progress.rs

```diff
@@ -0,0 +1,136 @@
+use core::mem;
+
+use crate::ui::{
+    component::{
+        base::ComponentExt,
+        paginated::Paginate,
+        text::paragraphs::{Paragraph, Paragraphs},
+        Child, Component, Event, EventCtx, Label, Never, Pad,
+    },
+    display::{self, Font},
+    geometry::{Insets, Rect},
+    model_tt::constant,
+    util::animation_disabled,
+};
+
+use super::theme;
+
+pub struct Progress<T, F>
+where
+    F: Fn(&str) -> T,
+{
+    title: Child<Label<T>>,
+    value: u16,
+    loader_y_offset: i16,
+    indeterminate: bool,
+    description: Child<Paragraphs<Paragraph<T>>>,
+    description_pad: Pad,
+    description_func: F,
```

Comment:
...but there _is_ an improvement we can do: change type of `description_func` from `F` to `fn(&str) -> Result<T, Error>` and pass in a static function (in our case `StrBuffer::alloc`)

and rename it to `update_description`? because "description func" was a function that _returns_ the description whereas this one will just allocate it

Reply:
Fixed in 489ddebfea522b4a339c9b1ef4e689978bcb1df1.

TIL that if you have a closure that doesn't capture any variables then it works with `fn(...) -> ...` syntax (not contained in this PR).